jvb/src/main/java/org/jitsi/videobridge/metrics/Metric.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright @ 2022 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.metrics;
+
+/**
+ * Supplies the current value of a metric, cast as an {@code Object}.
+ * Metrics are held in the {@link MetricsContainer}.
+ */
+@FunctionalInterface
+public interface Metric
+{
+    /**
+     * Supplies the value of a metric.
+     *
+     * @return the current value of a metric
+     */
+    Object getMetricValue();
```

Comment:
It crossed my mind; I agree with `Metric<T>`.
 I would no longer extend `GaugeMetric` for both `BooleanMetric` and `LongGaugeMetric`.
I would have to implement them concretely, since we'd have `GaugeMetric implements Metric<Double>`.
I could perhaps make `GaugeMetric` a `DoubleGaugeMetric`? Those would be used for metrics such as bitrate.

Does it make sense to skip the getters (such as `getLong()`) and just use the `Metric<T>` interface method? (less clutter)
```java
public final class LongGaugeMetric implements Metric<Long>
{
    private final Gauge gauge;
    // ...

    @Override
    public Long getMetricValue()
    {
        return (long) gauge.get();
    }
    // ...
}
```

Other operations _could_ also return the parameterized type, such as `Long incrementAndGet()`.
The only issue I see here is returning boxed types but I don't believe that's a problem.

Reply:
"Other operations could also return the parameterized type, such as Long incrementAndGet()."

I like that. You could implement LongGaugeMetric and BooleanGaugeMetric by delegating to a DoubleGaugeMetric, but I'm not sure it's worth the effort (compared to re-implementing the few methods they have).